src/nvim/api/vim.c

```diff
@@ -164,6 +165,39 @@ Object vim_eval(String str, Error *err)
   return rv;
 }
 
+/// Call the given function with the given arguments stored in an array.
+///
+/// @param functionName The function to call
+/// @param args the functions arguments packed in an Array
+/// @param[out] err Details of an error that may have occurred
+/// @return The result of the function call if it was successful
+Object vim_function(String functionName, Array args, Error *err)
+  FUNC_ATTR_DEFERRED
+{
+  Object rv = OBJECT_INIT;
+  typval_T vim_args[args.size+1];
```

Comment:
@splinterofchaos, @Pyrohh Comments there may say anything, VLAs are still not allowed: http://neovim.io/develop/style-guide.xml#Variable-Length_Arrays_and_alloca__. There were no modifications made to this entry.